File: php/services/donutService.php
<?php
class DonutService {
    private $donutRepository;

    public function __construct(DonutRepository $donutRepository) {
        $this->donutRepository = $donutRepository;
    }

    public function getAllDonuts(): array {
        return $this->donutRepository->findAll();
    }

    public function getDonutById(int $id): ?Donut {
        return $this->donutRepository->findById($id);
    }

    public function getDonutsByName(string $name): array {
        return $this->donutRepository->findByName($name);
    }

    public function getDonutsByCategory(string $category): array {
        return $this->donutRepository->findByCategory($category);
    }

    public function getDonutsByPrice(float $price): array {
        return $this->donutRepository->findByPrice($price);
    }
    public function getDonutsByIsNew(bool $isNew): array {
        return $this->donutRepository->findByIsNew($isNew);
    }
}

File: src/main/java/com/example/web/services/postgres/DonutService.java
package com.example.web.services.postgres;

import com.example.web.models.Donut;
import com.example.web.repositories.postgres.DonutRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class DonutService {
    private final DonutRepository donutRepository;
    @Autowired
    public DonutService(DonutRepository donutRepository) {
        this.donutRepository = donutRepository;
    }
    public List<Donut> findAll(){
        return donutRepository.findAll();
    }
    public Optional<Donut> findById(Integer id){
        return donutRepository.findById(id);
    }
    public Optional<Donut> findByName(String name){
        return donutRepository.findByName(name);
    }
    public List<Donut> findByCategory(String category){
        return donutRepository.findByCategory(category);
    }
    public List<Donut> findByPrice(Float price){
        return donutRepository.findByPrice(price);
    }
    public List<Donut> findByIsNew(Boolean isNew){
        return donutRepository.findByIsNew(isNew);
    }
}
